docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,314 @@
+   "source": [
+    "### Current output format\n",
+    "Currently `earthdata-varinfo` produces Python classes that can be used within Python scripts, Jupyter notebooks or a Python Read-Eval-Print Loop (REPL).\n",
+    "\n",
+    "`earthdata-varinfo` is also able to create UMM-Var compliant JSON records and publish these records to NASA's Common Metadata Repository (CMR)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example usage:\n",
+    "* [M2I1NXASM](https://cmr.uat.earthdata.nasa.gov/search/collections.umm_json?concept-id=C1256535511-EEDTEST)\n",
+    "* [GLDAS_NOAH10_3H](https://cmr.uat.earthdata.nasa.gov/search/collections.umm_json?concept-id=C1256543837-EEDTEST)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import CMR environment\n",
+    "from cmr import CMR_UAT\n",
+    "\n",
+    "# Import varinfo cmr_search functions\n",
+    "from varinfo.cmr_search import (get_granules, get_granule_link, \n",
+    "                                download_granule)\n",
+    "\n",
+    "# Import VarInfoFromNetCDF4 class and umm_var functions\n",
+    "# for reading and netCDF4 files and creating UMM-Var entries \n",
+    "from varinfo import VarInfoFromNetCDF4\n",
+    "\n",
+    "# Import functions for creating UMM-Var entry and\n",
+    "# publishing a UMM-Var entry\n",
+    "from varinfo.umm_var import (get_all_umm_var, get_umm_var, publish_all_umm_var,\n",
+    "                             publish_umm_var)\n",
+    "# Singular function call\n",
+    "from varinfo.generate_umm_var import generate_collection_umm_var"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Update authorization header to include your bearer token (e.g. 'Bearer TOKEN')\n",
+    "# or LaunchPad token (e.g. TOKEN)\n",
+    "auth_header = 'Header + Token or Only token'"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get granule response and granule download URL with ```get_granules``` and ```get_granule_link``` for collection **M2I1NXASM**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "granule_response = get_granules(concept_id='C1256535511-EEDTEST',\n",
+    "                                cmr_env=CMR_UAT,\n",
+    "                                auth_header=auth_header)\n",
+    "link = get_granule_link(granule_response)\n",
+    "link"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Then download the granule locally with `download_granule`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Saves granule to current directory or\n",
```

Comment:
This is a general comment that applies to a lot of the code-cells...

The comments in the code-cells are really informative, but there's quite a few places where I think the comments from the code could/should be migrated into markdown (and probably expanded a bit). That's particularly true for code-cells that are executing only one line of code.

My general thought here is that someone looking at a documentation notebook is probably going to start by reading the markdown sections first, rather than comments within code.

Reply:
*[ ] Make comes their own markdown cells if they are long and hold important info